master/getting-started/kubernetes/installation/migration-from-flannel.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Migrate a cluster using Flannel networking
+---
+
+# About
+
+The steps in this document explain how to migrate an existing Kubernetes cluster using Flannel networking to use Calico
+for networking.
+
+This is accomplished by installing a migration controller which performs a rolling update of the nodes in the cluster, executing
+the following steps:
+
+- Cordon and drain the node.
+- Migrate the node to use Calico.
+- Uncordon the node to allow workloads to be scheduled.
+
+At the end of you will have a fully-functional Calico cluster using VXLAN networking between pods.
+
+# Prerequisites
+
+- Migration requires a cluster using Flannel for networking.
+- Flannel must have been installed using a Kubernetes DaemonSet.
+- Flannel must be configured to use the Kubernetes API for storing its configuration (as opposed to etcd).
+- Requires flannel version v0.9.1 or greater (Canal version v3.7.0 or greater).
+
```

Comment:
Migration must not be used in a system which reconciles the CNI plugin?  e.g. aks-engine.  
@song-jiang thinks that very bad things will happen in that case...

Reply:
Yeah. In this case, the installer will keep trying to put Flannel back and may get some of Flannel pod back if unlucky.